gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -501,20 +503,15 @@ gst_tensor_time_sync_buffer_from_collectpad (GstCollectPads * collect,
       if (in_formats[i] != _NNS_TENSOR_FORMAT_FLEXIBLE) {
         GstTensorMetaInfo meta;
 
-        gst_tensor_info_convert_to_meta (&configs->info.info[i], &meta);
+        gst_tensor_info_convert_to_meta (_info, &meta);
         mem = gst_tensor_meta_info_append_header (&meta, in_mem[i]);
         gst_memory_unref (in_mem[i]);
       }
     }
 
-    if (i < NNS_TENSOR_SIZE_LIMIT) {
-      gst_buffer_append_memory (tensors_buf, mem);
-    } else {
-      GstTensorInfo *info = &configs->info.extra[i - NNS_TENSOR_SIZE_LIMIT];
-      if (!gst_tensor_buffer_append_memory (tensors_buf, mem, info)) {
-        nns_loge ("Failed to append memory to buffer.");
-        return FALSE;
-      }
+    if (!gst_tensor_buffer_append_memory (tensors_buf, mem, _info)) {
+      nns_loge ("Failed to append memory to buffer.");
+      return FALSE;
```

Comment:
need to call gst_memory_unref (_info) somewhere?

Reply:
I will add next PR after merging this,